src/test/kotlin/com/pubnub/api/presence/PresenceTest.kt

```diff
@@ -94,20 +97,70 @@ internal class PresenceTest {
         assertThat(presence, Matchers.isA(PresenceNoOp::class.java))
     }
 
+    @Test
+    fun `Leave events not created when suppressLeaveEvents is true and heartbeat interval is 0`() {
+        // given
+        val presence = Presence.create(
+            listenerManager = listenerManager, heartbeatInterval = 0.seconds, suppressLeaveEvents = true,
+            leaveProvider = object : LeaveProvider {
```

Comment:
We could use mock of LeaveProvider like this:

```
    @Test
    fun `Leave events not created when suppressLeaveEvents is true and heartbeat interval is 0`() {
        val leaveProviderMock: LeaveProvider = mockk()
        // given
        val presence = Presence.create(
            listenerManager = listenerManager, heartbeatInterval = 0.seconds, suppressLeaveEvents = true,
            leaveProvider = leaveProviderMock
        )

        // when
        presence.joined(setOf("abc"))
        presence.leftAll()

        // then
        verify(exactly = 0) { leaveProviderMock.getLeaveRemoteAction(any(), any()) }
    }
```

What do you think?

Reply:
done